api/controllers/AuthController.js

```diff
@@ -25,49 +25,67 @@ module.exports = {
      * might fail to complete the login and then be confused when they get redirected somewhere unexpected the next time they visit the site. */
     var login_info = {type: req.query.loginType, redirect: req.query.redirect || '/', validation: crypto.randomBytes(32).toString('hex')};
     req.session.validation = login_info.validation;
-    var auth_data = {state: JSON.stringify(login_info), duration: 'permanent', failureRedirect: '/login'};
```

Comment:
Was the failure redirect actually doing anything? I was always getting redirected to the 403 page on failure anyway.